specs/mgt-search.md

```diff
@@ -0,0 +1,182 @@
+| suggestion-input | null: no data | The template to render search input box. |
+| suggested-people-header | null: no data | The template to render people entity label. |
+| suggested-query-header | null: no data | The template to render text entity label. |
+| suggested-file-header | null: no data | The template to render file entity label. |
+| customized-{other entity type}-header | null: no data | The template to render other entity label, for example customized-sample1-header |
+| suggested-people | suggestionPeople[]: The people suggestion details  | The template to render people entity. |
+| suggested-query | suggestionQuery[]: The text suggestion details  | The template to render text entity. |
+| suggested-file | suggestionFile[]: The file suggestion details  | The template to render file entity. |
+| customized-{other entity type} | any[]: Customizing suggestion details  | The template to render customized entity, for example customized-sample1 |
+
+The following examples shows how to use customized template `suggested-query`.
+
+```html
+<mgt-search>
+    <template data-type="suggested-query">
+        <div>{{query}}</div>
+    </template>
+</mgt-search>
+```
+
+
+## Customized
+Customized entity types must be supported by microsoft graph suggestion API.
+```html
+<mgt-search entity-types="query,file,people,sample1">
+    <template data-type="customized-sample1-header">
+        <div>header hahaha</div>
+    </template>
+
+    <template data-type="customized-sample1">
+        <div>{{this}}</div>
+    </template>
+</mgt-search>
+```
+
+## Proposed Solution
+
+## Example
+
+### Example 1: basic usage
+```<mgt-search></mgt-search>```
+
+### Example 2: get suggestion data and define click and enter key action by adding a listener
+```
+<mgt-search> </mgt-search>
+
+<script>
+
+document.querySelector('mgt-search').addEventListener('onEntityClick', e => {
+    console.log(e.detail);
+});
+
+document.querySelector('mgt-search').addEventListener('onEnterPress', e => {
+  console.log(e.detail);
+});
+
+</script>
+
+```
+
+### Example4: Select entity types & Sort & Suggestion Amount
+It allows customized entity order and max suggestion account by using entity-types properties:
+For example below shows
+Order: query, file, people
+Amount: 2,1,3 ( 3 is default value)
+```html
+  <mgt-search entity-types="query-2, file-1, people"></mgt-search>
+```
+
+```html
+  <mgt-search entity-types="query-2, people"></mgt-search>
+```
+
+```html
+  <mgt-search entity-types="query-2, people, sample1-4"></mgt-search>
+```
+
+## Events
+
+The following events are fired from the component.
+
+| Event | Description |
+| --- | --- |
+| `onEntityClick` | When click an entity , the listener will be trigged. the suggested value of the clicked item as a parameter|
+| `onEnterPress` | When press enter key, the listener will be trigged, it has originalValue( input box value) and suggestedValue (suggestion) as parameters|
```

Comment:
Consider changing this to `querySubmitted` instead

```suggestion
| `querySubmitted` | When press enter key, the listener will be trigged, it has originalValue( input box value) and suggestedValue (suggestion) as parameters|
```

Reply:
changed to querySubmitted